storage/database/src/database.rs

```diff
@@ -35,6 +33,9 @@ Each iteration of the iterator has the potential to error as well."
 /// - <https://github.com/Cuprate/cuprate/pull/102#discussion_r1548695610>
 /// - <https://github.com/Cuprate/cuprate/pull/104>
 pub trait DatabaseIter<T: Table> {
+    /*
+    FIXME: <https://github.com/Cuprate/cuprate/issues/348>
```

Comment:
Would it be worth leaving this as-is then where used in `cuprate_blockchain` it branches on the backend? E.g.:

```rust
let tx_blobs = if cfg!(feature = "heed") {
    (first_tx_idx..(usize_to_u64(numb_non_miner_txs) + first_tx_idx))
        .map(|idx| {
            let tx_blob = tables.tx_blobs().get(&idx)?.0;

            Ok(Bytes::from(tx_blob))
        })
        .collect::<Result<_, RuntimeError>>()?
} else {
    tables
        .tx_blobs_iter()
        .get_range(first_tx_idx..(usize_to_u64(numb_non_miner_txs) + first_tx_idx))?
        .map(|tx_blob| Ok(Bytes::from(tx_blob?.0)))
        .collect::<Result<_, RuntimeError>>()?
};
```
Or is it not that important for now?

Reply:
Or actually, the `DatabaseIter::get_range` impl for `heed` would just use `get` internally.

Reply:
> Or actually, the DatabaseIter::get_range impl for heed would just use get internally.

I tried to do this but it's harder than it sounds. In `get_range` we don't know how much of the range we actually have, i.e if we ask for `0..100` but we only have the key `50` stored we still need to check the other 99 keys, not impossible but defiantly a footgun IMO. 

> Would it be worth leaving this as-is then where used in cuprate_blockchain it branches on the backend?

I don't think the complexity is worth it. This is only used for other nodes syncing from us, which from some initial tests, is still significantly faster than `monerod`